ragna/source_storages/_demo.py

```diff
@@ -30,8 +30,9 @@ def display_name(cls) -> str:
 
     def __init__(self) -> None:
         self._storage: list[dict[str, Any]] = []
+        self._corpus_name = "Demo Corpus"
```

Comment:
Shouldn't `_storage` become `dict[str, list[dict[str, Any]]]`, i.e. mapping the corpus name to the data we store? And to simplify stuff, I would actually use `Optional[str]` as key to also handle the default `corpus_name=None` case.